dbfit-java/derby/src/main/java/dbfit/environment/DerbyEnvironment.java

```diff
@@ -84,7 +85,80 @@ public Pattern getParameterPattern() {
 
     public Map<String, DbParameterAccessor> getAllProcedureParameters(
             String procName) throws SQLException {
-        throw new UnsupportedOperationException();
+        String[] qualifiers = procName.split("\\.");
+        String schemaName, procedureName;
+        if (qualifiers.length == 2) {
+            schemaName = qualifiers[0];
+            procedureName = qualifiers[1];
+        } else {
+            schemaName = getConnection().getSchema();
+            procedureName = qualifiers[0];
+        }
+        procedureName = procedureName.toUpperCase();
+        DatabaseMetaData dbmd = getConnection().getMetaData();
+        ResultSet rs;
+        rs = dbmd.getProcedures(null, schemaName, procedureName);
+        boolean isFunction = false;
+        if (rs.next()) {
+            rs.close();
+            rs = dbmd.getProcedureColumns(null, schemaName, procedureName, "%");
+        } else {
+            rs.close();
+            rs = dbmd.getFunctions(null, schemaName, procedureName);
+            if (rs.next()) {
+                isFunction = true;
+                rs.close();
+                rs = dbmd.getFunctionColumns(null, schemaName, procedureName, "%");
+            }
+        }
+        Map<String, DbParameterAccessor> allParams = new HashMap<String, DbParameterAccessor>();
```

Comment:
Yes. I didn't realise this was acceptable but compiler seems to infer the specific types for the generic constructor from the variable definition.